### PR TITLE
Revert "Drop support for julia 1.0-1.3 (#221)"

### PR DESCRIPTION
**Pull request by lgoettgens**

### CODE REVIEW

**Comment by lgoettgens** [comment]
As already expected, `Test = "1.0" fails on julia 1.0-1.3 ([log](https://github.com/JuliaTesting/Aqua.jl/actions/runs/6664533436?pr=224))

**Comment by lgoettgens** [comment]
`Test = "<0.0.1, 1"` indeed seems to work. Thanks @DilumAluthge!

**Comment by lgoettgens** [comment]
@fingolfin What do you think? Should we enable compatibility with julia 1.0 again (by merging this PR)? If yes, I should look through the changes in detail, because there were a few conflicts when reverting stuff.

If you approve this, could you change the required CI jobs back to needing 1.0-1.3 for ubuntu, 1.0 for windows and macOS, and instead remove 1.4 for windows and macOS?

**Comment by hyrodium** [comment]
I believe it is important that the code written for Julia 1.0 works with the latest versions of Julia and some packages. On the other hand, I think it is rare for users who do not upgrade their Julia version to only upgrade the version of Aqua.jl, so I feel there is no need to actively support Julia versions prior to 1.6 (LTS).

Sorry for repeating my previous comment (https://github.com/JuliaTesting/Aqua.jl/issues/216#issuecomment-1766572091), but I feel it would be nice to reduce the costs to maintain this package.

**Review by fingolfin — APPROVED** [review]
Honestly I don't think support for the older Julia versions is worth the hassle.

However, if you are willing to take care of the burden, I certainly won't object! In the worst case, we could still re-remove it later.

As such, I think you should make the final decision, Lars. I won't complain either way :-)

File: src/persistent_tasks.jl [hunk 1]
```diff
@@ -61,6 +61,8 @@ end
 In more complex cases, you may need to set up independently-callable functions
 to launch the tasks and set conditions that allow them to cleanly exit.
 
+On julia version 1.9 and before, this test always succeeds.
```
Comment (fingolfin) [inline src/persistent_tasks.jl:64]:
```suggestion
On Julia version 1.9 and before, this test always succeeds.
```

**Pull request merged**